File: design-patterns/command/ruby/delete_file.rb
require_relative 'command'

class DeleteFile < Command
  attr_reader :path
  
  def initialize(path)
    super "Delete file: #{path}"
    @path = path
  end
  
  def execute
    @contents = File.read(path) if File.exists?(path)
    File.delete(@path)
  end
  
  def undo
    File.open(path, 'w') do |f|
      f << @contents
    end
  end
end

File: design-patterns/java/DesignPatterns/src/ar/edu/unlam/cdp/patterns/command/DeleteFile.java
package ar.edu.unlam.cdp.patterns.command;

import java.io.File;
import java.io.IOException;
import java.nio.charset.Charset;

import com.google.common.io.Files;

public class DeleteFile extends Command {

	private String path;
	private String previousContents;

	DeleteFile(String description, String path) {
		super(description);
		this.path = path;
	}

	@Override
	public void execute() {

		try {
			
			File target = new File(this.path);

			if (target.exists()) {
				this.previousContents = Files.toString(target, Charset.defaultCharset());
			}

			target.delete();
			
		} catch (IOException e) {
			throw new RuntimeException(e);
		}
	}

	@Override
	public void undo() {
		try {
			
			File target = new File(this.path);
			Files.write(this.previousContents, target, Charset.defaultCharset());
			
		} catch (IOException e) {
			throw new RuntimeException(e);
		}
	}

}
